sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -1013,118 +1014,120 @@ class SubquerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
   }
 
   test("SPARK-23957 Remove redundant sort from subquery plan(in subquery)") {
-    withTempView("t1", "t2", "t3") {
-      Seq((1, 1), (2, 2)).toDF("c1", "c2").createOrReplaceTempView("t1")
-      Seq((1, 1), (2, 2)).toDF("c1", "c2").createOrReplaceTempView("t2")
-      Seq((1, 1, 1), (2, 2, 2)).toDF("c1", "c2", "c3").createOrReplaceTempView("t3")
-
-      // Simple order by
-      val query1 =
-        """
-           |SELECT c1 FROM t1
-           |WHERE
-           |c1 IN (SELECT c1 FROM t2 ORDER BY c1)
+    withSQLConf(SQLConf.OPTIMIZER_EXCLUDED_RULES.key -> RemoveNoopUnion.ruleName) {
```

Comment:
what happens to this test?

Reply:
Similar case, `Union` is removed and a check of number of `Sort` is failed. I think we can add a simple Filter as other test.